feat(consent)!: migrate to UMP SDK (#82)
invertase/react-native-google-mobile-ads: docs/european-user-consent.mdx

npx react-native run-android
```

### App Tracking Transparency

If you want to handle Apple's App Tracking Transparency requirements, ensure you've configured and published your [ATT message](https://developers.google.com/admob/ump/ios/quick-start#app_tracking_transparency) in your Google AdMob account.
Also, within your projects `app.json` file, you have to use the `user_tracking_usage_description` to to describe your usage:

```json
{
  "react-native-google-mobile-ads": {
    "android_app_id": "ca-app-pub-xxxxxxxx~xxxxxxxx",
    "ios_app_id": "ca-app-pub-xxxxxxxx~xxxxxxxx",
    "user_tracking_usage_description": "This identifier will be used to deliver personalized ads to you."
  }
}
```

Once set, rebuild your application.

### Requesting consent information

It is recommended you request consent information each time your application starts to determine if the consent modal should be shown, e.g. due to provider changes.

> The `AdsConsent` helper provides a promised based method to return the users consent status called `requestInfoUpdate`.

Reviewer: tiny grammar nitpick

```suggestion
The `AdsConsent` helper provides a promise based method to return the users consent status called `requestInfoUpdate`.
```

```js
import { AdsConsent } from 'react-native-google-mobile-ads';

const consentInfo = await AdsConsent.requestInfoUpdate();
```

> The result of the method returns an `AdsConsentInfo` interface, which provides information about consent form availability and the users consent status:

Reviewer: ```suggestion
The method returns an `AdsConsentInfo` interface, which provides information about consent form availability and the users consent status:
```

  - `NOT_REQUIRED`: User consent not required.
  - `OBTAINED`: User consent already obtained.
- **isConsentFormAvailable**: A boolean value. If `true` a consent form is available.

### Gathering user consent

To request consent, call these methods as early as possible within your app before presenting any ads.
Once the user has selected their preference, the `showForm` method returns an `AdsConsentFormResult` interface, containing the updated consent status:

```js
import { AdsConsent, AdsConsentStatus } from 'react-native-google-mobile-ads';

const consentInfo = await AdsConsent.requestInfoUpdate();

if (
  consentInfo.isConsentFormAvailable &&
  consentInfo.status === AdsConsentStatus.REQUIRED
) {
  const { status } = await AdsConsent.showForm();
}
```

> Do not persist the status. You could however store this locally in application state (e.g. React Context) and update the status on every app launch as it may change.

### Testing

When developing the consent flow, the behavior of the `AdsConsent` responses may not be reliable due to the environment
(e.g. using an emulator vs real device). It is possible to set a debug location to test the various responses from the
UMP SDK.

> If using a real device, ensure you whitelist it by passing the device ID, which can be obtained from native logs or using a library

Reviewer: ```suggestion
If using a real device, ensure you add it to the list of allowed devices by passing the device ID, which can be obtained from native logs or using a library
```

```js
import { AdsConsent, AdsConsentDebugGeography } from 'react-native-google-mobile-ads';

const consentInfo = await AdsConsent.requestInfoUpdate({
  debugGeography: AdsConsentDebugGeography.EEA,
  testDeviceIdentifiers: ['TEST-DEVICE-HASHED-ID'],
});
```

> It is also possible to reset the UMP state for test devices. This does not reset the ATT state though.

Reviewer: > This does not reset the ATT state though

Hmm - any pointers for inexperienced devs here? How does one reset the ATT state? Maybe a link to an apple doc or similar...
